### PR TITLE
Script for generating API documentation

### PR DESCRIPTION
**Pull request by jsancio**
- [x] Split the documentation into 3 files: package, service, capabilities
- [x] Fix the generation script to generate all the documentation for all of the 3 files.
- [x] Fix the CI save an artifact for all of the generated files
- [x] Fix the release CI to copy all of the files to S3
- [x] Fix the snapshot CI to copy all of the files to S3

### CODE REVIEW

File: build/generate_api.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+REPO_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+TARGET_DIR="$REPO_DIR/cosmos-server/target"
+mkdir -p "$TARGET_DIR"
+
+# Build a docker image with raml2html
+sudo docker image build --tag ramltools "$REPO_DIR/docker/ramltools"
+
+# Generate the API HTML using the built docker image
+sudo docker container run --volume=$REPO_DIR:/src --rm ramltools /usr/local/bin/raml2html \
```
Comment (cruhland) [inline build/generate_api.sh]: Put quotes around this argument: `"--volume=$REPO_DIR:/src"`

File: build/generate_api.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  /src/cosmos-server/src/main/resources/com/mesosphere/cosmos/handler/api.raml \
+  > "$TARGET_DIR/api.html"
+
+# Generate the API Swagger using the built docker image
+sudo docker container run --volume=$REPO_DIR:/src --rm ramltools /usr/local/bin/raml2swagger \
```
Comment (cruhland) [inline build/generate_api.sh]: Put quotes around this argument? `"--volume=$REPO_DIR:/src"`

File: docker/ramltools/raml2swagger.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+var transformer = require('/usr/local/lib/node_modules/api-spec-transformer');
+
+var ramlToSwagger = new transformer.Converter(
+    transformer.Formats.RAML10,
+    transformer.Formats.SWAGGER
+);
+
+var source = process.argv[2];
```
Comment (cruhland) [inline docker/ramltools/raml2swagger.js]: Why is this `argv[2]`? What's in `argv[0] and argv[1]`?

Comment (jsancio) [inline docker/ramltools/raml2swagger.js]:
```
argv[0] == nodejs
argv[1] == raml2swagger
```

I'll write a comment explaining that.

File: build/generate_api.sh (hunk 1 above)
Comment (jsancio) [inline build/generate_api.sh]: Done. Quoted the variable.

File: build/generate_api.sh (hunk 2 above)
Comment (jsancio) [inline build/generate_api.sh]: Done. Quoted the variable.

File: docker/ramltools/raml2swagger.js (hunk 3 above)
Comment (jsancio) [inline docker/ramltools/raml2swagger.js]: Done.

**Comment by jsancio** [comment]
@cruhland @larioj We are now generation 3 (service, package, capabilities) x 3 (html, raml, swagger) files.

https://teamcity.mesosphere.io/viewLog.html?buildId=593959&buildTypeId=DcosIo_Cosmos_PublishSnapshot&tab=buildLog#_state=7277,7244&focus=7261

**Pull request merged**